common/archiver/filestore/visibilityArchiver.go

```diff
@@ -336,11 +336,9 @@ func convertToExecutionInfo(record *archiverproto.ArchiveVisibilityRequest) *wor
 		Type: &commonpb.WorkflowType{
 			Name: record.WorkflowTypeName,
 		},
-		StartTime: &types.Int64Value{
-			Value: record.StartTimestamp},
-		ExecutionTime: record.ExecutionTimestamp,
-		CloseTime: &types.Int64Value{
-			Value: record.CloseTimestamp},
+		StartTime:     timestamp.TimePtr(timestamp.UnixOrZeroTime(record.StartTimestamp)),
+		ExecutionTime: timestamp.TimePtr(timestamp.UnixOrZeroTime(record.ExecutionTimestamp)),
+		CloseTime:     timestamp.TimePtr(timestamp.UnixOrZeroTime(record.CloseTimestamp)),
```

Comment:
`timestamp.UnixOrZeroTime` returns unix time if passed timestamp is not zero and `time.Time{}` if it is. The next goal is to remove this conversion.